build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts

```diff
@@ -19,26 +20,12 @@ kotlin {
 
     jvm()
 
-    sourceSets {
-        val nativeMain by creating {
-            dependsOn(commonMain.get())
-        }
-        val nativeTest by creating {
-            dependsOn(commonTest.get())
-        }
-        for (sourceSet in nativeTargets) {
-            getByName("${sourceSet}Main") {
-                dependsOn(nativeMain)
-            }
-            getByName("${sourceSet}Test") {
-                dependsOn(nativeTest)
-            }
-        }
+    @OptIn(ExperimentalKotlinGradlePluginApi::class)
+    targetHierarchy.default()
```

Comment:
This is a really cool feature that sets up the source sets the way you'd expect them to be (e.g. `native` is set up), and it is extensible.

It is experimental, so I can remove it if you want.

Reply:
kotlin multiplatform is experimental anyway so it's fine. Maybe try a quick `./gradlew publichToMavenLocal` to make sure there isn't any issues with publishing with the new layout.

Reply:
LGTM although I'm building on Linux so only the linuxX64 target gets built for native.

Reply:
seems good on macos too